### DISCUSSION TITLE
Get Grid Value

### DISCUSSION THREAD

**Discussion by agallics** [serenity-is/Serenity#7162] [Help]
Hi, I need to read the values of a column of a gridview (those visible after the filtered) to pass them as a reference to an endpoint
Is there a JS function that returns all the values of a column of a grid?

**Comment by VictorTomaili** [comment]
In the Grid class, you can use `this.getColumns()`.

**Reply by agallics** [reply]
Sorry, I did not explain it correctly.

I need the current list of values in the rows, not columns. I need to get the current filtered values into the table ignoring the paging (if there are 3 pages, I need to extract the values of all 3 pages).
I use quickfilters and HeaderFiltersMixin both.

**Reply by VictorTomaili** [reply]
Call the service with same filter without pagination parameters. 
If you need to show the grid without pagination, you can set `protected override usePager() { return false; }`

**Reply by agallics** [reply]
I managed a call to the service, but in that way I have to extract all QuickFIlters (that are a lot) using JQuery (Are there other way to do it?) and to extract the HeaderFiltersMixin selected by the user (I don't know how to do it) and than modify the call to the service. For more, it should make an extra call to the service.

Actually I wrote something like this
```
    protected getButtons(){
        var buttons = super.getButtons();
        buttons.push({
            title: "Extract data",
            onClick: _ => this.extractData(),
        });
        return buttons;
    }

    private extractData() {
        MyRowService.List(
            {
                // My filters here
            },
            response => console.log(response),
        );
    }
```
Is there is a better way?

What I need is just to extract the filtered data set actually viewable (ignoring pagination) in the grid instead. I just need to access that data and than make a Ajax call. I don't want to remove pagination. Is it possible?

EDIT:

I was able to extract the content in the grid with this command `this.getGrid().getData().getItems()`, but return only the items in the current page, I want also the ones in the other pages

**Reply by VictorTomaili** [reply]
`this.view.params` eliminate the pagination from the params.
